search_and_download.py

```diff
@@ -0,0 +1,81 @@
+# (C) British Crown Copyright 2017, Met Office
+"""
+A command-line utility to hit the API endpoint with specific search parameters
+in the query string and download all the available files
+
+    Args:
+        api-key: The Api-key to access the data
+
+"""
+import argparse
+import os.path
+import time
+
+import requests
+
+
+START_TIME_STEP = 12
+END_TIME_STEP = 36
+DOWNLOAD_DIR = 'downloaded_objects'
+API_GATEWAY_URL = "https://api.metoffice.gov.uk/objects/v1/global/data"
+
+
+def perform_search(api_key):
+    # The model reference time for the current date
+    current_date = time.strftime("20%y-%m-%d")
+    forecast_reference_time = current_date + "T00:00:00Z"
+
+    # The search parameters to include for diagnostic names
+    name = ("rainfall_rate",
+            "air_pressure_at_sea_level",
+            "wind_speed",
+            "wind_from_direction")
+
+    # The time step search parameters (start and end timesteps in seconds)
+    start_time = START_TIME_STEP*3600
+    end_time = END_TIME_STEP*3600
+
+    # Loop through parameters to find and download available data
+    for name_parameter in name:
+        for forecast_period in range(start_time, end_time, 3600):
```

Comment:
actually 3600 is used alot. Maybe it should be a constant?

`HOUR_AS_SECONDS = 3600`

Reply:
I would leave this as is so that it clearly shows a range() is being used to iterate over. This is core python and should be self explanatory to an experienced developer with no knowledge of how models work.

Reply:
Disagree.